src/main/java/com/feedzai/commons/sql/abstraction/engine/AbstractTranslator.java

```diff
@@ -263,8 +263,8 @@ public String translate(K k) {
         if (o != null) {
             if (!k.isQuote()) {
                 result = o.toString();
-            } else if (o instanceof String) {
-                result = singleQuotize(escapeSql((String) o));
+            } else if (o instanceof String || o instanceof Enum) {
+                result = singleQuotize(escapeSql(String.valueOf(o)));
```

Comment:
Agree!

On one hand, we can override `toString` and write to DB what we want.
On the other hand, `toString` can be just a human readable description which we may want to change while keeping the DB value the same.

Because of this last case and more generally because the value in the DB normally doesn't matter much as long as it matches the enum value, I'll change to `name`